### PR TITLE
Add support for nuget, pypi, and sourceforge in url2purl

### PR DESCRIPTION
**Pull request by tdruez**

### CODE REVIEW

File: src/packageurl/contrib/url2purl.py [hunk 1]
```diff
@@ -37,6 +38,9 @@
     from urllib.parse import urlparse  # Python 3
     from urllib.parse import unquote_plus
 
+from pip._internal.wheel import Wheel
```
Comment (pombredanne) [inline src/packageurl/contrib/url2purl.py]: This is all internal and not API... and the library may not have been installed with pip.
How could you document this dependency? Should this is an extra in setup.py? and what if the API changes? Could this be vendored here instead?

Comment (tdruez) [reply src/packageurl/contrib/url2purl.py]: @pombredanne I've remove the dependency on pip internal in the latest commit.

Comment (pombredanne) [reply src/packageurl/contrib/url2purl.py]: Thanks!

**Comment by pombredanne** [comment]
LGTM merging

**Pull request merged**